### PR TITLE
Making TagHelperContent implement IHtmlContent.

### PR DESCRIPTION
**Pull request by sornaks**

### CODE REVIEW

File: src/Microsoft.AspNet.Razor.Runtime/project.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
         "Microsoft.Framework.BufferEntryCollection.Sources": { "type": "build", "version": "1.0.0-*" },
         "Microsoft.Framework.ClosedGenericMatcher.Sources": { "type": "build", "version": "1.0.0-*" },
         "Microsoft.Framework.CopyOnWriteDictionary.Sources": { "type": "build", "version": "1.0.0-*" },
-        "Microsoft.Framework.NotNullAttribute.Sources": { "type": "build", "version": "1.0.0-*" }
+        "Microsoft.Framework.NotNullAttribute.Sources": { "type": "build", "version": "1.0.0-*" },
+        "Microsoft.AspNet.Html.Abstractions": "1.0.0-*"
```
Comment (rynowak) [inline src/Microsoft.AspNet.Razor.Runtime/project.json]: Sort these

File: test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/DefaultTagHelperContentTest.cs [hunk 2]
```diff
@@ -622,5 +624,21 @@ public void Fluent_Clear_Set_Append_WritesExpectedContent()
...
+        public void WriteTo_WritesToATextWriter()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var writer = new StringWriter();
+            tagHelperContent.SetContent("Hello ");
+            tagHelperContent.Append("World");
+
+            // Act
+            tagHelperContent.WriteTo(writer, new HtmlEncoder());
```
Comment (rynowak) [inline test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/DefaultTagHelperContentTest.cs]: Always use the test encoders. Even if it makes no difference. Someone will have to look at this later and go ???

**Comment by rynowak** [comment]
:shipit: after fixing minor comments

**Comment by sornaks** [comment]
Thanks! https://github.com/aspnet/Razor/commit/ec5d1b4f4a83b0faf8cfe2556dfc8bde4efeaeb9

File: src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs [hunk 3]
```diff
@@ -208,5 +210,14 @@ public override IEnumerator<string> GetEnumerator()
             // can use this to iterate through the values of the buffer.
             return _buffer.GetEnumerator();
         }
+
+        /// <inheritdoc />
+        public override void WriteTo(TextWriter writer, IHtmlEncoder encoder)
+        {
+            foreach (var entry in _buffer)
+            {
+                writer.Write(entry);
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs]: How come the encoder isn't being used?

Comment (sornaks) [inline src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs]: Currently, this function receives encoded content only. I've an upcoming PR where I delay the encoding till the content is written.

**Pull request closed without merging**